spec/index.bs

```diff
@@ -4168,6 +4168,98 @@ enum GPUStoreOp {
 };
 </script>
 
+## Rasterization state ## {#render-pass-encoder-rasterization-state}
+
+The render pass encoder has several methods which affect how draw commands are rasterized to the
+attachments used by this encoder.
+
+### <dfn method for=GPURenderPassEncoder>setViewport(x, y, width, height, minDepth, maxDepth)</dfn> ### {#GPURenderPassEncoder-setViewport}
+
+<div algorithm="GPURenderPassEncoder.setViewport">
+    <strong>this:</strong> of type {{GPURenderPassEncoder}}.
+
+    **Arguments:**
+        - `float` x - Left side of the viewport in pixels
+        - `float` y - Top side of the viewport in pixels
+        - `float` |width| of the viewport in pixels
+        - `float` |height| of the viewport in pixels
+        - `float` |minDepth| - Minimum depth value of the viewport
+        - `float` |maxDepth| - Maximum depth value of the viewport. May be less than or equal to
+            |minDepth|.
+
+    **Returns:** void
+
+    Sets the viewport used during the rasterization stage to linearly map from normalized device
+    coordinates to viewport coordinates.
+
+    <div class=validusage dfn-for=GPURenderPassEncoder.setViewport>
+        <dfn abstract-op>setViewport Valid Usage</dfn>
+
```

Comment:
Dawn's source had a comment indicating that the validation for `x` and `y` was uncertain, and that it may need to restrict negative numbers. The source itself didn't check for that, though, so I've left it off here. Happy to add it if that's the intent.

Reply:
Vulkan has constraints where the viewport has to stay in a [3*width, 3*height] box centered around the default viewport, but it's pretty complicated. We might want something simpler for WebGPU.

Also can you add an Issue: that it should be allowed for GPUs to use fixed point for viewport coordinates (or even round them to integers)?

Reply:
Added the issue.